theano/tensor/nnet/nnet.py

```diff
@@ -401,6 +396,11 @@ def make_node(self, dy, sm, axis=-1):
             dy = tensor.shape_padleft(dy, n_ones=1)
         if sm.ndim == 1:
             sm = tensor.shape_padleft(sm, n_ones=1)
+        # Check if axis is a scalar
+        try:
+            opt.get_scalar_constant_value(axis)
+        except tensor.NotScalarConstantError:
+            raise ValueError('axis must be a scalar. Use abstract_softmax instead Got ', axis.type)
         return Apply(self, [dy, sm, axis], [sm.type()])
```

Comment:
Sorry, but I still don't get why `axis` is a symbolic input node at all. `theano.tensor.elemwise.Sum` has a nonsymbolic tuple of axes:
```
>>> import theano.tensor as T
>>> x = T.matrix('x')
>>> y = x.sum(axis=-1)
>>> y.owner.op.axis
(1,)
>>> y.owner.inputs
[x]
```
What's the advantage of making it symbolic?